elasticsearch/bulk/bulk.go

```diff
@@ -439,9 +450,9 @@ func getBytes(batchItems []BatchItem) [][]byte {
 }
 
 func getActions(batchItems []BatchItem) []*document.ESActionDocument {
-	batchActions := make([]*document.ESActionDocument, 0, len(batchItems))
-	for _, batchItem := range batchItems {
-		batchActions = append(batchActions, batchItem.Action)
+	result := make([]*document.ESActionDocument, len(batchItems))
```

Comment:
for range with assigning raw indexes has better performance so I refactored here, I also added test.